AppleCider/models/hyrax_models/spectranet.py

```diff
@@ -0,0 +1,204 @@
+    blocks = []
+    for i in range(depth):
+        blocks.append(
+            SpectraNetBlock(
+                in_channels=in_channel if i == 0 else out_channel * k,
+                out_channels=out_channel,
+                kernel_sizes=kernel_sizes,
+                use_ln=use_ln,
+                do_pool=(do_pool if i == depth - 1 else False),
+            )
+        )
+    return nn.Sequential(*blocks), k
+
+
+@hyrax_model
+class SpectraNet(nn.Module):
+    """Model for training spectra data.
+    Can be configured to output either a transient
+    classification or give a redshift score prediction
+    (see the "AppleCider/default_config.toml" file).
+    """
+
+    def __init__(self, config=None, data_sample=None):
+        super().__init__()
+
+        self.config = config
+
+        spectranet_config = config["model"]["SpectraNet"]
+
+        self.redshift = spectranet_config["redshift"]
+        kernel_sizes_per_stage = spectranet_config["kernel_sizes_per_stage"]
+        depths = spectranet_config["depths"]
+        use_ln_stages = spectranet_config["use_ln_stages"]
+        channels = spectranet_config["channels"]
+        flat_dim = spectranet_config["flat_dim"]
+        class_order = spectranet_config["class_order"]
+
+        if (
+            len(depths)
+            != len(use_ln_stages)
+            != len(channels)
+            != len(kernel_sizes_per_stage)
+        ):
+            raise ValueError(
+                "depths, use_ln_stages, channels, and kernel_sizes_per_stage must be the same length."
+            )
+
+        # create each layer
+        self.stages = []
+        self.ks = []
+        for i in range(len(depths)):
+            in_channel = 1 if i == 0 else channels[i - 1]
+            out_channel = channels[i]
+            depth = depths[i]
+            kernel_size = kernel_sizes_per_stage[i]
+            use_ln_stage = use_ln_stages[i]
+            do_pool = False if i == len(depths) - 1 else True
+
+            stage, k = make_stage(
+                in_c=in_channel,
+                out_c=out_channel,
+                depth=depth,
+                kernel_sizes=kernel_size,
+                use_ln=use_ln_stage,
+                do_pool=do_pool,
+            )
+
+            self.stages.append(stage)
+            self.ks.append(k)
+
+        # create the sequence for all layers
+        self.all_stages = nn.Sequential(*self.stages)
+
+        if self.redshift:
+            self.regressor = nn.Sequential(
+                nn.Linear(flat_dim, 384),
+                nn.LayerNorm(384),
+                nn.GELU(),
+                nn.Dropout(0.5),
+                nn.Linear(384, 1),
+            )
+        else:
+            self.classifier = nn.Sequential(
+                nn.Linear(flat_dim, 384),
+                nn.LayerNorm(384),
+                nn.GELU(),
+                nn.Dropout(0.5),
+                nn.Linear(384, class_order),
+            )
+
+    def forward(self, x):
```

Comment:
I believe that `forward` will always receive data that has passed through `to_tensor` whether that is for training or inference. The difference is that `train_step` will call the `forward` method during training, but during inference, the data is fed directly from `to_tensor` to `forward`. 

Long preamble to say that forward should accept the batch as input, and then unpack it like what happens on line 183 in `train_step`, but it won't use `labels` or `redshifts`...

i.e. 
```
def forward(self, batch):
    inputs, _, _ = batch
    # keep the rest the same
```